### PR TITLE
feature/book-and-label-classes-methods

### PR DESCRIPTION
**Pull request by ClaudiaRojasSoto**
# Description:
This pull request meets all the following requirements:
## `books.rb`:
- [ ] Create a `Book `class in a separate .rb file.
- [ ] All `Book` class properties visible in the diagram should be defined and set up in the constructor method.
## `labels.rb`:
- [ ] Create a `Label `class with an association to the Item class (in a separate .rb file).
- [ ] All `Label` class properties visible in the diagram should be defined and set up in the constructor method.

##  **_Implement methods_**:

1. **`add_item` method in the `Label` class**

   - [ ] Should take an instance of the `Item` class as an input
   - [ ] Should add the input item to the collection of items
   - [ ] Should add self as a property of the item object (by using the correct setter from the item object)

2. `**can_be_archived?()**` in the `Book` class

   - [ ] Should override the method from the parent class
   - [ ] Should return true if the parent's method returns true OR if `cover_state` equals to "**bad**"
   - [ ] Otherwise, it should return false

Please @PabloBona and @francksefu review my feature and give me some feedback to improve my work. Thank you in advance. 
Greetings,
@ClaudiaRojasSoto

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require_relative 'item'
+
+# Class representing a Book
+class Book < Item
+  attr_accessor :publisher, :cover_state, :label
```
Comment (francksefu) [inline book.rb:5]: As  : label  (as attr_accessor) is already attr_accessor in Item, it is not important to make it  again  as attr_accessor in the child file

Comment (francksefu) [reply book.rb:5]: [Optional]

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  def initialize(params = {})
+    super(params)
+    @publisher = params[:publisher]
+    @cover_state = params[:cover_state]
+    @label = params[:label]
+  end
```
Comment (francksefu) [inline book.rb:12]: [Optional] For 3 parameters, it's good to use no-object in the middle but do this : 
def initialize(publisher, cover_state, label) 
when you define

**Review by francksefu — APPROVED** [review]
@francksefu  approved this pull request.

Hi @ClaudiaRojasSoto ,

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

Highlights
Add unit tests for all the methods in all the classes (✅)) 👏
All test files are placed in a different directory 👏
Test has passed 👏 👏
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  def can_be_archived?
+    super || @cover_state == 'bad'
+  end
```
Comment (francksefu) [inline book.rb:16]: This method must be private

**Review by francksefu — CHANGES_REQUESTED** [review]

File: book.rb (hunk 3 above)
Comment (ClaudiaRojasSoto) [reply book.rb:17]: I Changed the method to private. Thank you for your feedback

**Review by francksefu — APPROVED** [review]
You did very great job @ClaudiaRojasSoto  👍  Your project is Approved 🥇

**Pull request merged**